test/y2storage/boot_requirements_errors_test.rb

```diff
@@ -67,13 +69,19 @@
   end
 
   describe "#errors" do
+    RSpec.shared_examples "no errors" do
+      it "does not contain an error" do
+        expect(checker.warnings).to be_empty
+      end
+    end
+
     RSpec.shared_examples "missing boot partition" do
       it "contains an error for missing boot partition" do
         expect(checker.warnings.size).to eq(1)
         expect(checker.warnings).to all(be_a(Y2Storage::SetupError))
 
-        missing_volume = checker.warnings.first.missing_volume
-        expect(missing_volume).to eq(boot_volume)
+        message = checker.warnings.first.message
+        expect(message).to match(/Missing device for \/boot/)
```

Comment:
Now we have a new class `VolumeSpecificationBuilder` [1]. Maybe it would be better to check that there is a warning for the missing volume:

```
boot_volume = Y2Storage::VolumeSpecificationBuilder.new.for("/boot")
expect(checker.warnings).to include(an_object_having_attributes(missing_volume: boot_volume))
```

The same for the rest of test.

[1] https://github.com/yast/yast-storage-ng/blob/master/src/lib/y2storage/volume_specification_builder.rb#L58

Reply:
well, even when I really like reusability and DRY in production code, for tests I prefer more explicit checks which is trivial, so we are sure tests are correct.